### PR TITLE
[runtime env] Improve working dir messaging

### PR DESCRIPTION
**Pull request by architkulkarni**
## Why are these changes needed?
Currently, nothing is printed while the working directory is uploaded.  When the working directory is large or the upload is slow, this makes it look like `ray.init(address="ray://...")` is hanging.  

This PR prints "Uploading..." and "Complete" when the working directory is larger than a given threshold (For now, it's 1 MiB).  The tradeoff here is that we don't want spammy output in the common case, which is a small working directory consisting of source code files.

(If an *individual file* is greater than 10 MiB we already print a warning; this PR doesn't change that.)

- This PR also sets the log level in the Ray Client worker to match the log level in Ray, which is INFO by default, as well as the log format.  Preexisting client INFO logs have been changed to DEBUG or ERROR according to their severity.
- 
Sample output:
```
❯ ipython mytest.py
2021-10-26 11:33:28,873 WARNING packaging.py:93 -- File /Users/archit/code/test_wd_log/output.dat is very large (50.0 MiB). Consider adding this file to the 'excludes' list to skip uploading it: `ray.init(..., runtime_env={'excludes': ['/Users/archit/code/test_wd_log/output.dat']})`
2021-10-26 11:33:29,053 INFO packaging.py:204 -- Pushing large local file package (50.0 MiB) to Ray cluster...
2021-10-26 11:33:30,079 INFO packaging.py:207 -- Pushed local files.
```

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/_private/runtime_env/working_dir.py [hunk 1]
```diff
@@ -159,14 +164,27 @@ def match(p: Path):
...
         raise RuntimeError(
-            "working_dir package exceeds the maximum size of 100MiB. You "
-            "can exclude large files using the 'excludes' option to the "
-            "runtime_env.")
+            f"working_dir package has size {file_size_str}, which exceeds the "
+            "maximum size of 100 MiB. You can exclude large files using the "
+            "'excludes' field of the runtime_env.")
+
+    if file_size > SILENT_UPLOAD_SIZE_THRESHOLD:
+        logger.warning(f"Pushing local files to cluster ({file_size_str})...")
```
Comment (architkulkarni) [inline python/ray/_private/runtime_env/working_dir.py:181]: I'm not sure `logger.warning` is the right way to go here.  I tried `logger.info` and the following happened:

- When using Ray client, the logs didn't appear.
- When not using Ray client, the log did appear: `2021-09-24 15:51:25,101 INFO working_dir.py:428 -- Pushing local files to cluster (50.00 MiB)...`

In both cases, the `logger` kwarg isn't being passed, so it's just using `default_logger = logging.getLogger(__name__)` and I'm not sure why there's a discrepancy.

Comment (architkulkarni) [reply python/ray/_private/runtime_env/working_dir.py:181]: With `logger.warning`, we get the following:
 - When using Ray client, we get a log line `Uploading local files (99.0 MiB)...`
 - When not using Ray client, we get a formatted log line: `2021-09-24 17:21:18,947 WARNING working_dir.py:181 -- Pushing local files to cluster (2.3 MiB)...`
 
We don't want that `WARNING` to show up, so I'll have to figure out what's going on here

**Review by edoakes — APPROVED** [review]
Looks good, the warning is indeed pretty weird though...

**Comment by edoakes** [comment]
@architkulkarni can you do a bit more investigation and try to figure out how to log without the `WARNING` label?

**Review by edoakes — COMMENTED** [review]
Looks good, just want to make sure we don't spam with ray client log messages.

**Review by edoakes — APPROVED** [review]

**Pull request merged**